diff -ur /prefix-portage-2.2.01.15553/bin/ebuild.sh /prefix-portage-2.2.01.15553/bin/ebuild.sh
--- /prefix-portage-2.2.01.15553/bin/ebuild.sh	2010-09-23 21:23:22.000000000 +0400
+++ /prefix-portage-2.2.01.15553/bin/ebuild.sh	2010-09-23 21:27:48.000000000 +0400
@@ -882,7 +882,9 @@
 }
 
 exeopts() {
-	export EXEOPTIONS="$@"
+	EXEOPTIONS="$@"
+	[ -z "${EPREFIX}" ] || EXEOPTIONS=`echo "${EXEOPTIONS}" | sed 's! -[og] [[:alnum:]]*!!g'`
+	export EXEOPTIONS
 
 	# `install` should never be called with '-s' ...
 	hasq -s ${EXEOPTIONS} && die "Never call exeopts() with -s"
