diff -ur libdc1394-2.1.3/dc1394/bayer.c libdc1394-2.1.3/dc1394/bayer.c
--- libdc1394-2.1.3/dc1394/bayer.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/bayer.c	2011-07-09 17:03:02.000000000 +0400
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include "conversions.h"
+#include "platform.h"
 
 #define CLIP(in, out)\
    in = in < 0 ? 0 : in;\
@@ -1977,7 +1977,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_bayer_decoding_8bit(const uint8_t *restrict bayer, uint8_t *restrict rgb, uint32_t sx, uint32_t sy, dc1394color_filter_t tile, dc1394bayer_method_t method)
 {
     switch (method) {
@@ -2003,7 +2003,7 @@
 
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_bayer_decoding_16bit(const uint16_t *restrict bayer, uint16_t *restrict rgb, uint32_t sx, uint32_t sy, dc1394color_filter_t tile, dc1394bayer_method_t method, uint32_t bits)
 {
     switch (method) {
@@ -2118,7 +2118,7 @@
     return DC1394_MEMORY_ALLOCATION_FAILURE;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_debayer_frames(dc1394video_frame_t *in, dc1394video_frame_t *out, dc1394bayer_method_t method)
 {
     if ((method<DC1394_BAYER_METHOD_MIN)||(method>DC1394_BAYER_METHOD_MAX))
diff -ur libdc1394-2.1.3/dc1394/capture.c libdc1394-2.1.3/dc1394/capture.c
--- libdc1394-2.1.3/dc1394/capture.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/capture.c	2011-07-09 17:03:02.000000000 +0400
@@ -24,7 +24,7 @@
 #include "platform.h"
 #include "internal.h"
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_capture_setup (dc1394camera_t *camera, uint32_t num_dma_buffers,
         uint32_t flags)
 {
@@ -35,7 +35,7 @@
     return d->capture_setup (cpriv->pcam, num_dma_buffers, flags);
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_capture_stop (dc1394camera_t *camera)
 {
     dc1394camera_priv_t * cpriv = DC1394_CAMERA_PRIV (camera);
@@ -45,7 +45,7 @@
     return d->capture_stop (cpriv->pcam);
 }
 
-int
+DLL_PUBLIC int
 dc1394_capture_get_fileno (dc1394camera_t * camera)
 {
     dc1394camera_priv_t * cpriv = DC1394_CAMERA_PRIV (camera);
@@ -55,7 +55,7 @@
     return d->capture_get_fileno (cpriv->pcam);
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_capture_dequeue (dc1394camera_t * camera, dc1394capture_policy_t policy,
         dc1394video_frame_t **frame)
 {
@@ -66,7 +66,7 @@
     return d->capture_dequeue (cpriv->pcam, policy, frame);
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_capture_enqueue (dc1394camera_t * camera, dc1394video_frame_t * frame)
 {
     dc1394camera_priv_t * cpriv = DC1394_CAMERA_PRIV (camera);
@@ -76,7 +76,7 @@
     return d->capture_enqueue (cpriv->pcam, frame);
 }
 
-dc1394bool_t
+DLL_PUBLIC dc1394bool_t
 dc1394_capture_is_frame_corrupt (dc1394camera_t * camera,
         dc1394video_frame_t * frame)
 {
diff -ur libdc1394-2.1.3/dc1394/control.c libdc1394-2.1.3/dc1394/control.c
--- libdc1394-2.1.3/dc1394/control.c	2009-06-11 07:28:14.000000000 +0400
+++ libdc1394-2.1.3/dc1394/control.c	2011-07-09 17:03:02.000000000 +0400
@@ -28,7 +28,7 @@
 #include "internal.h"
 #include "offsets.h"
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_camera_set_broadcast(dc1394camera_t *camera, dc1394bool_t pwr)
 {
     dc1394camera_priv_t * priv = DC1394_CAMERA_PRIV (camera);
@@ -38,7 +38,7 @@
     return d->set_broadcast (priv->pcam, pwr);
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_camera_get_broadcast(dc1394camera_t *camera, dc1394bool_t *pwr)
 {
     dc1394camera_priv_t * priv = DC1394_CAMERA_PRIV (camera);
@@ -48,7 +48,7 @@
     return d->get_broadcast (priv->pcam, pwr);
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_reset_bus (dc1394camera_t * camera)
 {
     dc1394camera_priv_t * priv = DC1394_CAMERA_PRIV (camera);
@@ -58,7 +58,7 @@
     return d->reset_bus (priv->pcam);
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_read_cycle_timer (dc1394camera_t * camera,
                          uint32_t * cycle_timer, uint64_t * local_time)
 {
@@ -69,7 +69,7 @@
     return d->read_cycle_timer (priv->pcam, cycle_timer, local_time);
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_camera_get_node (dc1394camera_t *camera, uint32_t *node,
         uint32_t * generation)
 {
@@ -153,7 +153,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_camera_print_info(dc1394camera_t *camera, FILE* fd)
 {
     dc1394camera_priv_t * cpriv = DC1394_CAMERA_PRIV (camera);
@@ -197,7 +197,7 @@
  Collects the available features for the camera
  described by node and stores them in features.
 *****************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_all(dc1394camera_t *camera, dc1394featureset_t *features)
 {
     uint32_t i, j;
@@ -218,7 +218,7 @@
  Stores the bounds and options associated with the
  feature described by feature->id
 *****************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get(dc1394camera_t *camera, dc1394feature_info_t *feature)
 {
     uint64_t offset;
@@ -340,7 +340,7 @@
 
  Displays the bounds and options of the given feature
 *****************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_print(dc1394feature_info_t *f, FILE *fd)
 {
     int fid= f->id;
@@ -469,7 +469,7 @@
 
  Displays the entire feature set stored in features
 *****************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_print_all(dc1394featureset_t *features, FILE *fd)
 {
     uint32_t i, j;
@@ -492,7 +492,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_camera_reset(dc1394camera_t *camera)
 {
     dc1394error_t err;
@@ -501,7 +501,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_supported_modes(dc1394camera_t *camera, dc1394video_modes_t *modes)
 {
     dc1394error_t err;
@@ -579,7 +579,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_supported_framerates(dc1394camera_t *camera, dc1394video_mode_t video_mode, dc1394framerates_t *framerates)
 {
     dc1394framerate_t framerate;
@@ -624,7 +624,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_framerate(dc1394camera_t *camera, dc1394framerate_t *framerate)
 {
     uint32_t value;
@@ -638,7 +638,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_framerate(dc1394camera_t *camera, dc1394framerate_t framerate)
 {
     dc1394error_t err;
@@ -652,7 +652,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_mode(dc1394camera_t *camera, dc1394video_mode_t *mode)
 {
     dc1394error_t err;
@@ -691,7 +691,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_mode(dc1394camera_t *camera, dc1394video_mode_t  mode)
 {
     uint32_t format, min;
@@ -737,7 +737,7 @@
 
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_iso_speed(dc1394camera_t *camera, dc1394speed_t *speed)
 {
     dc1394error_t err;
@@ -767,7 +767,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_iso_speed(dc1394camera_t *camera, dc1394speed_t speed)
 {
     dc1394error_t err;
@@ -806,7 +806,7 @@
     return err;;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_iso_channel(dc1394camera_t *camera, uint32_t * channel)
 {
     dc1394error_t err;
@@ -827,7 +827,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_iso_channel(dc1394camera_t *camera, uint32_t channel)
 {
     dc1394error_t err;
@@ -865,7 +865,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_operation_mode(dc1394camera_t *camera, dc1394operation_mode_t  *mode)
 {
     dc1394error_t err;
@@ -888,7 +888,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_operation_mode(dc1394camera_t *camera, dc1394operation_mode_t  mode)
 {
     dc1394error_t err;
@@ -918,7 +918,7 @@
 
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_camera_set_power(dc1394camera_t *camera, dc1394switch_t pwr)
 {
     dc1394error_t err;
@@ -938,7 +938,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_transmission(dc1394camera_t *camera, dc1394switch_t pwr)
 {
     dc1394error_t err;
@@ -956,7 +956,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_transmission(dc1394camera_t *camera, dc1394switch_t *is_on)
 {
     dc1394error_t err;
@@ -968,7 +968,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_one_shot(dc1394camera_t *camera, dc1394switch_t pwr)
 {
     dc1394error_t err;
@@ -988,7 +988,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_one_shot(dc1394camera_t *camera, dc1394bool_t *is_on)
 {
     uint32_t value;
@@ -998,7 +998,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_multi_shot(dc1394camera_t *camera, dc1394bool_t *is_on, uint32_t *numFrames)
 {
     uint32_t value;
@@ -1010,7 +1010,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_set_multi_shot(dc1394camera_t *camera, uint32_t numFrames, dc1394switch_t pwr)
 {
     dc1394error_t err;
@@ -1030,7 +1030,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_whitebalance_get_value(dc1394camera_t *camera, uint32_t *u_b_value, uint32_t *v_r_value)
 {
     uint32_t value;
@@ -1042,7 +1042,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_whitebalance_set_value(dc1394camera_t *camera, uint32_t u_b_value, uint32_t v_r_value)
 {
     uint32_t curval;
@@ -1056,7 +1056,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_temperature_get_value(dc1394camera_t *camera, uint32_t *target_temperature, uint32_t *temperature)
 {
     uint32_t value;
@@ -1067,7 +1067,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_temperature_set_value(dc1394camera_t *camera, uint32_t target_temperature)
 {
     dc1394error_t err;
@@ -1083,7 +1083,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_whiteshading_get_value(dc1394camera_t *camera, uint32_t *r_value, uint32_t *g_value, uint32_t *b_value)
 {
     uint32_t value;
@@ -1097,7 +1097,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_whiteshading_set_value(dc1394camera_t *camera, uint32_t r_value, uint32_t g_value, uint32_t b_value)
 {
     uint32_t curval;
@@ -1114,7 +1114,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_get_mode(dc1394camera_t *camera, dc1394trigger_mode_t *mode)
 {
     uint32_t value;
@@ -1129,7 +1129,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_set_mode(dc1394camera_t *camera, dc1394trigger_mode_t mode)
 {
     dc1394error_t err;
@@ -1152,7 +1152,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_get_supported_sources(dc1394camera_t *camera, dc1394trigger_sources_t *sources)
 {
     uint32_t value;
@@ -1176,7 +1176,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_get_source(dc1394camera_t *camera, dc1394trigger_source_t *source)
 {
     uint32_t value;
@@ -1191,7 +1191,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_set_source(dc1394camera_t *camera, dc1394trigger_source_t source)
 {
     dc1394error_t err;
@@ -1213,7 +1213,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_value(dc1394camera_t *camera, dc1394feature_t feature, uint32_t *value)
 {
     uint32_t quadval;
@@ -1239,7 +1239,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_set_value(dc1394camera_t *camera, dc1394feature_t feature, uint32_t value)
 {
     uint32_t quadval;
@@ -1266,7 +1266,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_is_present(dc1394camera_t *camera, dc1394feature_t feature, dc1394bool_t *value)
 {
 /*
@@ -1352,7 +1352,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_is_readable(dc1394camera_t *camera, dc1394feature_t feature, dc1394bool_t *value)
 {
     dc1394error_t err;
@@ -1372,7 +1372,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_is_switchable(dc1394camera_t *camera, dc1394feature_t feature, dc1394bool_t *value)
 {
     dc1394error_t err;
@@ -1392,7 +1392,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_power(dc1394camera_t *camera, dc1394feature_t feature, dc1394switch_t *value)
 {
     dc1394error_t err;
@@ -1412,7 +1412,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_set_power(dc1394camera_t *camera, dc1394feature_t feature, dc1394switch_t value)
 {
     dc1394error_t err;
@@ -1442,7 +1442,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_modes(dc1394camera_t *camera, dc1394feature_t feature, dc1394feature_modes_t *modes)
 {
     dc1394error_t err;
@@ -1480,7 +1480,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_mode(dc1394camera_t *camera, dc1394feature_t feature, dc1394feature_mode_t *mode)
 {
     dc1394error_t err;
@@ -1513,7 +1513,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_set_mode(dc1394camera_t *camera, dc1394feature_t feature, dc1394feature_mode_t mode)
 {
     dc1394error_t err;
@@ -1554,7 +1554,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_boundaries(dc1394camera_t *camera, dc1394feature_t feature, uint32_t *min, uint32_t *max)
 {
     dc1394error_t err;
@@ -1582,7 +1582,7 @@
  * Memory load/save functions
  */
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_memory_busy(dc1394camera_t *camera, dc1394bool_t *value)
 {
     uint32_t quadlet;
@@ -1592,7 +1592,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_memory_save(dc1394camera_t *camera, uint32_t channel)
 {
     dc1394error_t err=dc1394_set_control_register(camera, REG_CAMERA_MEM_SAVE_CH, (uint32_t)((channel & 0xFUL) << 28));
@@ -1603,7 +1603,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_memory_load(dc1394camera_t *camera, uint32_t channel)
 {
     dc1394error_t err=dc1394_set_control_register(camera, REG_CAMERA_CUR_MEM_CH, (uint32_t)((channel & 0xFUL) << 28));
@@ -1615,7 +1615,7 @@
  * Trigger functions
  */
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_set_polarity(dc1394camera_t *camera, dc1394trigger_polarity_t polarity)
 {
     dc1394error_t err;
@@ -1633,7 +1633,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_get_polarity(dc1394camera_t *camera, dc1394trigger_polarity_t *polarity)
 {
     uint32_t value;
@@ -1644,7 +1644,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_has_polarity(dc1394camera_t *camera, dc1394bool_t *polarity)
 {
     dc1394error_t err;
@@ -1661,7 +1661,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_set_power(dc1394camera_t *camera, dc1394switch_t pwr)
 {
     dc1394error_t err=dc1394_feature_set_power(camera, DC1394_FEATURE_TRIGGER, pwr);
@@ -1669,7 +1669,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_external_trigger_get_power(dc1394camera_t *camera, dc1394switch_t *pwr)
 {
     dc1394error_t err=dc1394_feature_get_power(camera, DC1394_FEATURE_TRIGGER, pwr);
@@ -1677,7 +1677,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_software_trigger_set_power(dc1394camera_t *camera, dc1394switch_t pwr)
 {
     dc1394error_t err;
@@ -1692,7 +1692,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_software_trigger_get_power(dc1394camera_t *camera, dc1394switch_t *pwr)
 {
     uint32_t value;
@@ -1704,7 +1704,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_data_depth(dc1394camera_t *camera, uint32_t *depth)
 {
     dc1394error_t err;
@@ -1740,7 +1740,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_absolute_control(dc1394camera_t *camera, dc1394feature_t feature, dc1394switch_t *pwr)
 {
     dc1394error_t err;
@@ -1760,7 +1760,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_set_absolute_control(dc1394camera_t *camera, dc1394feature_t feature, dc1394switch_t pwr)
 {
     dc1394error_t err;
@@ -1790,7 +1790,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_has_absolute_control(dc1394camera_t *camera, dc1394feature_t feature, dc1394bool_t *value)
 {
     dc1394error_t err;
@@ -1811,7 +1811,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_video_get_bandwidth_usage(dc1394camera_t *camera, uint32_t *bandwidth)
 {
     uint32_t format, qpp;
@@ -1855,7 +1855,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_absolute_boundaries(dc1394camera_t *camera, dc1394feature_t feature, float *min, float *max)
 {
     dc1394error_t err=DC1394_SUCCESS;
@@ -1873,7 +1873,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_get_absolute_value(dc1394camera_t *camera, dc1394feature_t feature, float *value)
 {
     dc1394error_t err=DC1394_SUCCESS;
@@ -1888,7 +1888,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_feature_set_absolute_value(dc1394camera_t *camera, dc1394feature_t feature, float value)
 {
     dc1394error_t err=DC1394_SUCCESS;
@@ -1907,7 +1907,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_pio_set(dc1394camera_t *camera, uint32_t value)
 {
     dc1394error_t err=DC1394_SUCCESS;
@@ -1919,7 +1919,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_pio_get(dc1394camera_t *camera, uint32_t *value)
 {
     dc1394error_t err=DC1394_SUCCESS;
@@ -1938,7 +1938,7 @@
 /*
   Create a new dc1394 struct, which also initialises the library
 */
-dc1394_t *
+DLL_PUBLIC dc1394_t *
 dc1394_new (void)
 {
     dc1394_t * d = calloc (1, sizeof (dc1394_t));
@@ -1984,7 +1984,7 @@
 /*
   Free a dc1394 struct, which also terminates the use of the library
 */
-void
+DLL_PUBLIC void
 dc1394_free (dc1394_t * d)
 {
     free_enumeration (d);
@@ -2047,7 +2047,7 @@
     return str;
 }
 
-dc1394camera_t *
+DLL_PUBLIC dc1394camera_t *
 dc1394_camera_new_unit (dc1394_t * d, uint64_t guid, int unit)
 {
     int i;
@@ -2168,7 +2168,7 @@
     return NULL;
 }
 
-dc1394camera_t *
+DLL_PUBLIC dc1394camera_t *
 dc1394_camera_new (dc1394_t * d, uint64_t guid)
 {
     return dc1394_camera_new_unit (d, guid, -1);
@@ -2178,7 +2178,7 @@
   Free a camera structure as well as resources used by that camera (bandwidth,
   ISO channels, etc...)
 */
-void
+DLL_PUBLIC void
 dc1394_camera_free(dc1394camera_t *camera)
 {
     dc1394camera_priv_t * cpriv = DC1394_CAMERA_PRIV (camera);
diff -ur libdc1394-2.1.3/dc1394/conversions.c libdc1394-2.1.3/dc1394/conversions.c
--- libdc1394-2.1.3/dc1394/conversions.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/conversions.c	2011-07-09 17:03:02.000000000 +0400
@@ -23,6 +23,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include "conversions.h"
+#include "platform.h"
 
 // this should disappear...
 extern void swab();
@@ -542,7 +542,7 @@
 
 // change a 16bit stereo image (8bit/channel) into two 8bit images on top
 // of each other
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_deinterlace_stereo(uint8_t *restrict src, uint8_t *restrict dest, uint32_t width, uint32_t height)
 {
     register int i = (width*height)-1;
@@ -557,7 +557,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_convert_to_YUV422(uint8_t *src, uint8_t *dest, uint32_t width, uint32_t height, uint32_t byte_order,
                          dc1394color_coding_t source_coding, uint32_t bits)
 {
@@ -594,7 +594,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_convert_to_MONO8(uint8_t *restrict src, uint8_t *restrict dest, uint32_t width, uint32_t height, uint32_t byte_order,
                         dc1394color_coding_t source_coding, uint32_t bits)
 {
@@ -613,7 +613,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_convert_to_RGB8(uint8_t *restrict src, uint8_t *restrict dest, uint32_t width, uint32_t height, uint32_t byte_order,
                        dc1394color_coding_t source_coding, uint32_t bits)
 {
@@ -720,7 +720,7 @@
     return DC1394_MEMORY_ALLOCATION_FAILURE;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_convert_frames(dc1394video_frame_t *in, dc1394video_frame_t *out)
 {
 
@@ -967,7 +967,7 @@
 
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_deinterlace_stereo_frames(dc1394video_frame_t *in, dc1394video_frame_t *out, dc1394stereo_method_t method)
 {
     dc1394error_t err;
diff -ur libdc1394-2.1.3/dc1394/enumeration.c libdc1394-2.1.3/dc1394/enumeration.c
--- libdc1394-2.1.3/dc1394/enumeration.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/enumeration.c	2011-07-09 17:03:02.000000000 +0400
@@ -274,7 +274,7 @@
     return 0;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_camera_enumerate (dc1394_t * d, dc1394camera_list_t **list)
 {
     if (refresh_enumeration (d) < 0)
@@ -302,7 +302,7 @@
 /*
   Free a list of cameras returned by dc1394_enumerate_cameras()
  */
-void
+DLL_PUBLIC void
 dc1394_camera_free_list (dc1394camera_list_t *list)
 {
     if (list)
diff -ur libdc1394-2.1.3/dc1394/format7.c libdc1394-2.1.3/dc1394/format7.c
--- libdc1394-2.1.3/dc1394/format7.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/format7.c	2011-07-09 17:03:02.000000000 +0400
@@ -229,7 +229,7 @@
 /* External functions */
 /**********************/
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_max_image_size(dc1394camera_t *camera,
                                   dc1394video_mode_t video_mode,
                                   uint32_t *horizontal_size,
@@ -250,7 +250,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_unit_size(dc1394camera_t *camera,
                              dc1394video_mode_t video_mode,
                              uint32_t *horizontal_unit,
@@ -271,7 +271,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_image_position(dc1394camera_t *camera,
                                   dc1394video_mode_t video_mode,
                                   uint32_t *left_position,
@@ -293,7 +293,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_image_size(dc1394camera_t *camera,
                               dc1394video_mode_t video_mode,
                               uint32_t *width,
@@ -314,7 +314,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_color_coding(dc1394camera_t *camera,
                                 dc1394video_mode_t video_mode,
                                 dc1394color_coding_t *color_coding)
@@ -334,7 +334,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_color_codings(dc1394camera_t *camera,
                                  dc1394video_mode_t video_mode,
                                  dc1394color_codings_t *color_codings)
@@ -360,7 +360,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_pixel_number(dc1394camera_t *camera,
                                  dc1394video_mode_t video_mode,
                                 uint32_t *pixnum)
@@ -379,7 +379,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_total_bytes(dc1394camera_t *camera,
                                dc1394video_mode_t video_mode,
                                uint64_t *total_bytes)
@@ -406,7 +406,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_packet_parameters(dc1394camera_t *camera,
                                      dc1394video_mode_t video_mode, uint32_t *unit_bytes,
                                      uint32_t *max_bytes)
@@ -426,7 +426,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_packet_size(dc1394camera_t *camera,
                                dc1394video_mode_t video_mode,
                                uint32_t *packet_size)
@@ -449,7 +449,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_set_image_position(dc1394camera_t *camera,
                                   dc1394video_mode_t video_mode, uint32_t left,
                                   uint32_t top)
@@ -465,7 +465,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_set_image_size(dc1394camera_t *camera,
                               dc1394video_mode_t video_mode, uint32_t width,
                               uint32_t height)
@@ -481,7 +481,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_set_color_coding(dc1394camera_t *camera,
                                 dc1394video_mode_t video_mode, dc1394color_coding_t color_coding)
 {
@@ -496,7 +496,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_set_packet_size(dc1394camera_t *camera,
                                dc1394video_mode_t video_mode,
                                uint32_t packet_bytes)
@@ -511,7 +511,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_recommended_packet_size(dc1394camera_t *camera,
                                            dc1394video_mode_t video_mode, uint32_t *packet_size)
 {
@@ -529,7 +529,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_packets_per_frame(dc1394camera_t *camera,
                                      dc1394video_mode_t video_mode, uint32_t *ppf)
 {
@@ -574,7 +574,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_unit_position(dc1394camera_t *camera,
                                  dc1394video_mode_t video_mode,
                                  uint32_t *horizontal_pos,
@@ -602,7 +602,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_frame_interval(dc1394camera_t *camera, dc1394video_mode_t video_mode, float *interval)
 {
     dc1394error_t err;
@@ -619,7 +619,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_data_depth(dc1394camera_t *camera, dc1394video_mode_t video_mode, uint32_t *data_depth)
 {
     dc1394error_t err;
@@ -652,7 +652,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_color_filter(dc1394camera_t *camera, dc1394video_mode_t video_mode, dc1394color_filter_t *color_filter)
 {
     dc1394error_t err;
@@ -671,7 +671,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_mode_info(dc1394camera_t *camera, dc1394video_mode_t video_mode, dc1394format7mode_t *f7_mode)
 {
     dc1394error_t err=DC1394_SUCCESS;
@@ -737,7 +737,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_modeset(dc1394camera_t *camera, dc1394format7modeset_t *info)
 {
     dc1394error_t err;
@@ -762,7 +762,7 @@
     return err;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_set_roi(dc1394camera_t *camera,
                        dc1394video_mode_t video_mode,
                        dc1394color_coding_t color_coding,
@@ -932,7 +932,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_format7_get_roi(dc1394camera_t *camera,
                        dc1394video_mode_t video_mode,
                        dc1394color_coding_t *color_coding,
diff -ur libdc1394-2.1.3/dc1394/log.c libdc1394-2.1.3/dc1394/log.c
--- libdc1394-2.1.3/dc1394/log.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/log.c	2011-07-09 17:03:02.000000000 +0400
@@ -24,6 +24,7 @@
 #include "log.h"
 #include <stdarg.h>
 #include <stdlib.h>
+#include "platform.h"
 
 static void
 default_errorlog_handler(dc1394log_t type, const char *message, void* user)
@@ -59,7 +59,7 @@
 static void *warninglog_data = NULL;
 static void *debuglog_data = NULL;
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_log_register_handler(dc1394log_t type, void(*log_handler)(dc1394log_t type, const char *message, void* user), void* user) {
     switch (type) {
     case DC1394_LOG_ERROR:
@@ -79,7 +79,7 @@
     }
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_log_set_default_handler(dc1394log_t type) {
     switch (type) {
     case DC1394_LOG_ERROR:
@@ -100,7 +100,7 @@
 }
 
 
-void dc1394_log_error(const char *format,...)
+DLL_PUBLIC void dc1394_log_error(const char *format,...)
 {
     char string[1024];
     if (system_errorlog_handler != NULL) {
@@ -112,7 +112,7 @@
 }
 
 
-void dc1394_log_warning(const char *format,...)
+DLL_PUBLIC void dc1394_log_warning(const char *format,...)
 {
     char string[1024];
     if (system_warninglog_handler != NULL) {
@@ -124,7 +124,7 @@
 }
 
 
-void dc1394_log_debug(const char *format,...)
+DLL_PUBLIC void dc1394_log_debug(const char *format,...)
 {
     char string[1024];
     if (system_debuglog_handler != NULL) {
diff -ur libdc1394-2.1.3/dc1394/platform.h libdc1394-2.1.3/dc1394/platform.h
--- libdc1394-2.1.3/dc1394/platform.h	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/platform.h	2011-07-09 17:30:35.000000000 +0400
@@ -26,6 +26,32 @@
 #include <stdint.h>
 #include <dc1394/camera.h>
 
+/* http://gcc.gnu.org/wiki/Visibility */
+#if defined _WIN32 || defined __CYGWIN__
+  #ifdef BUILDING_DLL
+    #ifdef __GNUC__
+      #define DLL_PUBLIC __attribute__ ((dllexport))
+    #else
+      #define DLL_PUBLIC __declspec(dllexport)
+    #endif
+  #else
+    #ifdef __GNUC__
+      #define DLL_PUBLIC __attribute__ ((dllimport))
+    #else
+      #define DLL_PUBLIC __declspec(dllimport)
+    #endif
+  #endif
+  #define DLL_LOCAL
+#else
+  #if __GNUC__ >= 4
+    #define DLL_PUBLIC __attribute__ ((visibility ("default")))
+    #define DLL_LOCAL  __attribute__ ((visibility ("hidden")))
+  #else
+    #define DLL_PUBLIC
+    #define DLL_LOCAL
+  #endif
+#endif
+
 typedef struct _platform_t platform_t;
 typedef struct _platform_device_t platform_device_t;
 typedef struct _platform_camera_t platform_camera_t;
diff -ur libdc1394-2.1.3/dc1394/utils.c libdc1394-2.1.3/dc1394/utils.c
--- libdc1394-2.1.3/dc1394/utils.c	2009-06-11 07:28:14.000000000 +0400
+++ libdc1394-2.1.3/dc1394/utils.c	2011-07-09 17:03:02.000000000 +0400
@@ -90,7 +90,7 @@
     "Basler error: unknown SFF chunk"
 };
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_get_image_size_from_video_mode(dc1394camera_t *camera, dc1394video_mode_t video_mode, uint32_t *w, uint32_t *h)
 {
     dc1394error_t err;
@@ -160,7 +160,7 @@
 }
 
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_framerate_as_float(dc1394framerate_t framerate_enum, float *framerate)
 {
     switch(framerate_enum)  {
@@ -192,7 +192,7 @@
     return DC1394_INVALID_FRAMERATE;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_is_color(dc1394color_coding_t color_coding, dc1394bool_t *is_color)
 {
     switch(color_coding)  {
@@ -215,7 +215,7 @@
     return DC1394_INVALID_COLOR_CODING;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_get_color_coding_data_depth(dc1394color_coding_t color_coding, uint32_t * bits)
 {
     switch(color_coding) {
@@ -239,7 +239,7 @@
     return DC1394_INVALID_COLOR_CODING;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_get_color_coding_bit_size(dc1394color_coding_t color_coding, uint32_t* bits)
 {
     switch(color_coding) {
@@ -268,7 +268,7 @@
     return DC1394_INVALID_COLOR_CODING;
 }
 
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_get_color_coding_from_video_mode(dc1394camera_t *camera, dc1394video_mode_t video_mode, dc1394color_coding_t *color_coding)
 {
     dc1394error_t err;
@@ -334,25 +334,25 @@
     return DC1394_FAILURE;
 }
 
-dc1394bool_t
+DLL_PUBLIC dc1394bool_t
 dc1394_is_video_mode_scalable(dc1394video_mode_t video_mode)
 {
     return ((video_mode>=DC1394_VIDEO_MODE_FORMAT7_MIN)&&(video_mode<=DC1394_VIDEO_MODE_FORMAT7_MAX));
 }
 
-dc1394bool_t
+DLL_PUBLIC dc1394bool_t
 dc1394_is_video_mode_still_image(dc1394video_mode_t video_mode)
 {
     return ((video_mode>=DC1394_VIDEO_MODE_FORMAT6_MIN)&&(video_mode<=DC1394_VIDEO_MODE_FORMAT6_MAX));
 }
 
-dc1394bool_t
+DLL_PUBLIC dc1394bool_t
 dc1394_is_same_camera(dc1394camera_id_t id1, dc1394camera_id_t id2)
 {
     return ((id1.guid==id2.guid)&&(id1.unit==id2.unit));
 }
 
-const char *
+DLL_PUBLIC const char *
 dc1394_feature_get_string(dc1394feature_t feature)
 {
     if ((feature>DC1394_FEATURE_MAX)||(feature<DC1394_FEATURE_MIN))
@@ -361,7 +361,7 @@
     return dc1394_feature_desc[feature-DC1394_FEATURE_MIN];
 }
 
-const char *
+DLL_PUBLIC const char *
 dc1394_error_get_string(dc1394error_t error)
 {
 
@@ -380,7 +380,7 @@
  * Version : 16/02/2005
  */
 
-uint16_t
+DLL_PUBLIC DLL_PUBLIC uint16_t
 dc1394_checksum_crc16 (const uint8_t* buffer, uint32_t buffer_size)
 {
     uint32_t i, j, c, bit;
diff -ur libdc1394-2.1.3/dc1394/vendor/avt.c libdc1394-2.1.3/dc1394/vendor/avt.c
--- libdc1394-2.1.3/dc1394/vendor/avt.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/vendor/avt.c	2011-07-09 17:30:52.000000000 +0400
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include "vendor/avt.h"
+#include "platform.h"
 
 /********************************************************/
 /* Configuration Register Offsets for Advances features */
@@ -96,7 +97,7 @@
 /************************************************************************/
 /* Get Version          (Read Only)                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_version(dc1394camera_t *camera,
                        uint32_t *UCType, uint32_t *Version,
                        uint32_t *Camera_ID, uint32_t *FPGA_Version)
@@ -127,7 +128,7 @@
 /************************************************************************/
 /* Get Advanced feature inquiry                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_advanced_feature_inquiry(dc1394camera_t *camera,
                                         dc1394_avt_adv_feature_info_t *adv_feature)
 {
@@ -207,7 +208,7 @@
 /************************************************************************/
 /* Print Advanced features                                                 */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_print_advanced_feature(dc1394_avt_adv_feature_info_t *adv_feature)
 {
 
@@ -271,7 +272,7 @@
 /************************************************************************/
 /* Get shading mode                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_shading(dc1394camera_t *camera,
                        dc1394bool_t *on_off, dc1394bool_t *compute,
                        dc1394bool_t *show, uint32_t *frame_nb)
@@ -307,7 +308,7 @@
 /************************************************************************/
 /* Set shading mode                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_shading(dc1394camera_t *camera,
                        dc1394bool_t on_off, dc1394bool_t compute,
                        dc1394bool_t show, uint32_t frame_nb)
@@ -343,7 +344,7 @@
 /************************************************************************/
 /* Get shading  mem ctrl                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_shading_mem_ctrl(dc1394camera_t *camera, dc1394bool_t *en_write,
                                 dc1394bool_t *en_read, uint32_t *addroffset)
 {
@@ -373,7 +374,7 @@
 /************************************************************************/
 /* Set shading mem ctrl                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_shading_mem_ctrl(dc1394camera_t *camera,
                                 dc1394bool_t en_write, dc1394bool_t en_read, uint32_t addroffset)
 {
@@ -403,7 +404,7 @@
 /************************************************************************/
 /* Get shading  info                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_shading_info(dc1394camera_t *camera, uint32_t *MaxImageSize)
 {
     dc1394error_t err;
@@ -424,7 +425,7 @@
 /************************************************************************/
 /* Get Multiple slope parameters        (HDR)                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_multiple_slope(dc1394camera_t *camera,
                               dc1394bool_t *on_off, uint32_t *points_nb,uint32_t *kneepoint1,
                               uint32_t *kneepoint2, uint32_t *kneepoint3)
@@ -458,7 +459,7 @@
 /************************************************************************/
 /* Set Multiple slope parameters                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_multiple_slope(dc1394camera_t *camera,
                               dc1394bool_t on_off, uint32_t points_nb, uint32_t kneepoint1,
                               uint32_t kneepoint2, uint32_t kneepoint3)
@@ -495,7 +496,7 @@
 /************************************************************************/
 /* Get Shutter Timebase                                                 */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_timebase(dc1394camera_t *camera, uint32_t *timebase_id)
 {
     dc1394error_t err;
@@ -515,7 +516,7 @@
 /************************************************************************/
 /* Set Shutter Timebase (acquisition must be stopped)                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_timebase(dc1394camera_t *camera, uint32_t timebase_id)
 {
     dc1394error_t err;
@@ -538,7 +539,7 @@
 /************************************************************************/
 /* Get Extented Shutter                                                  */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_extented_shutter(dc1394camera_t *camera, uint32_t *timebase_id)
 {
     dc1394error_t err;
@@ -559,7 +560,7 @@
 /************************************************************************/
 /* Set Extented shutter                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_extented_shutter(dc1394camera_t *camera, uint32_t timebase_id)
 {
     dc1394error_t err;
@@ -583,7 +584,7 @@
 /************************************************************************/
 /* Get MaxResolution          (Read Only)                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_MaxResolution(dc1394camera_t *camera, uint32_t *MaxHeight, uint32_t *MaxWidth)
 {
     dc1394error_t err;
@@ -605,7 +606,7 @@
 /************************************************************************/
 /* Get Auto Shutter                                                          */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_auto_shutter(dc1394camera_t *camera, uint32_t *MinValue, uint32_t *MaxValue)
 {
     dc1394error_t err;
@@ -630,7 +631,7 @@
 /************************************************************************/
 /* Set Auto shutter                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_auto_shutter(dc1394camera_t *camera, uint32_t MinValue, uint32_t MaxValue)
 {
     dc1394error_t err;
@@ -649,7 +650,7 @@
 /************************************************************************/
 /* Get Auto Gain                                                          */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_auto_gain(dc1394camera_t *camera, uint32_t *MinValue, uint32_t *MaxValue)
 {
     dc1394error_t err;
@@ -671,7 +672,7 @@
 /************************************************************************/
 /* Set Auto gain                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_auto_gain(dc1394camera_t *camera, uint32_t MinValue, uint32_t MaxValue)
 {
     dc1394error_t err;
@@ -691,7 +692,7 @@
 /************************************************************************/
 /* Get Trigger delay                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_trigger_delay(dc1394camera_t *camera, dc1394bool_t *on_off, uint32_t *DelayTime)
 {
     dc1394error_t err;
@@ -714,7 +715,7 @@
 /************************************************************************/
 /* Set Trigger delay                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_trigger_delay(dc1394camera_t *camera, dc1394bool_t on_off, uint32_t DelayTime)
 {
     dc1394error_t err;
@@ -741,7 +742,7 @@
 /************************************************************************/
 /* Get Mirror                                                                 */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_mirror(dc1394camera_t *camera, dc1394bool_t *on_off)
 {
     dc1394error_t err;
@@ -760,7 +761,7 @@
 /************************************************************************/
 /* Set Mirror                                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_mirror(dc1394camera_t *camera, dc1394bool_t on_off)
 {
     dc1394error_t err;
@@ -779,7 +780,7 @@
 /************************************************************************/
 /* Get DSNU                                                                 */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_dsnu(dc1394camera_t *camera, dc1394bool_t *on_off,uint32_t *frame_nb)
 {
     dc1394error_t err;
@@ -800,7 +801,7 @@
 /************************************************************************/
 /* Set DSNU                                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_dsnu(dc1394camera_t *camera,
                     dc1394bool_t on_off, dc1394bool_t compute, uint32_t frame_nb)
 {
@@ -838,7 +839,7 @@
 /************************************************************************/
 /* Get BLEMISH                                                                 */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_blemish(dc1394camera_t *camera, dc1394bool_t *on_off, uint32_t *frame_nb)
 {
     dc1394error_t err;
@@ -859,7 +860,7 @@
 /************************************************************************/
 /* Set BLEMISH                                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_blemish(dc1394camera_t *camera,
                        dc1394bool_t on_off, dc1394bool_t compute, uint32_t frame_nb)
 {
@@ -900,7 +901,7 @@
 /************************************************************************/
 /* Get IO   REG_CAMERA_AVT_IO_INP_CTRLx        or REG_CAMERA_AVT_IO_OUTP_CTRLx        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_io(dc1394camera_t *camera, uint32_t IO,
                   dc1394bool_t *polarity, uint32_t *mode, dc1394bool_t *pinstate)
 {
@@ -926,7 +927,7 @@
 /************************************************************************/
 /* Set IO   REG_CAMERA_AVT_IO_INP_CTRLx        or REG_CAMERA_AVT_IO_OUTP_CTRLx        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_io(dc1394camera_t *camera,uint32_t IO,
                   dc1394bool_t polarity, uint32_t mode, dc1394bool_t pinstate)
 {
@@ -957,7 +958,7 @@
 /************************************************************************/
 /* BusReset IEEE1394                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_reset(dc1394camera_t *camera)
 {
     dc1394error_t err;
@@ -975,7 +976,7 @@
 /************************************************************************/
 /* Get Lookup Tables (LUT)                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_lut(dc1394camera_t *camera, dc1394bool_t *on_off, uint32_t *lutnb)
 {
     dc1394error_t err;
@@ -998,7 +999,7 @@
 /************************************************************************/
 /* Set Lookup Tables (LUT)                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_lut(dc1394camera_t *camera, dc1394bool_t on_off, uint32_t lutnb)
 {
     dc1394error_t err;
@@ -1025,7 +1026,7 @@
 /************************************************************************/
 /* Get LUT ctrl                                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_lut_mem_ctrl(dc1394camera_t *camera, dc1394bool_t *en_write,
                             uint32_t * AccessLutNo,uint32_t *addroffset)
 {
@@ -1052,7 +1053,7 @@
 /************************************************************************/
 /* Set LUT ctrl                                                                 */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_lut_mem_ctrl(dc1394camera_t *camera,
                             dc1394bool_t en_write, uint32_t AccessLutNo, uint32_t addroffset)
 {
@@ -1083,7 +1084,7 @@
 /************************************************************************/
 /* Get LUT  info                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_lut_info(dc1394camera_t *camera, uint32_t *NumOfLuts, uint32_t *MaxLutSize)
 {
     dc1394error_t err;
@@ -1106,7 +1107,7 @@
 /************************************************************************/
 /* Get Automatic white balance        with Area Of Interest AOI                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_aoi(dc1394camera_t *camera,
                    dc1394bool_t *on_off, int *left, int *top, int *width, int *height)
 {
@@ -1144,7 +1145,7 @@
 /************************************************************************/
 /* Set Automatic white balance with Area Of Interest AOI                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_aoi(dc1394camera_t *camera,
                    dc1394bool_t on_off,int left, int top, int width, int height)
 {
@@ -1172,7 +1173,7 @@
 /************************************************************************/
 /* Get test_images                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_test_images(dc1394camera_t *camera, uint32_t *image_no)
 {
     dc1394error_t err;
@@ -1192,7 +1193,7 @@
 /************************************************************************/
 /* Set test_images                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_test_images(dc1394camera_t *camera, uint32_t image_no)
 {
     dc1394error_t err;
@@ -1216,7 +1217,7 @@
 /************************************************************************/
 /* Get frame info                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_frame_info(dc1394camera_t *camera, uint32_t *framecounter)
 {
     dc1394error_t err;
@@ -1236,7 +1237,7 @@
 /************************************************************************/
 /* Reset frame info                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_reset_frame_info(dc1394camera_t *camera)
 {
     dc1394error_t err;
@@ -1250,7 +1251,7 @@
 /************************************************************************/
 /* Get Deferred image transport                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_deferred_trans(dc1394camera_t *camera,
                               dc1394bool_t *HoldImage, dc1394bool_t * FastCapture,
                               uint32_t *FifoSize, uint32_t *NumOfImages )
@@ -1281,7 +1282,7 @@
 /************************************************************************/
 /* Set Deferred image transport                                                */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_set_deferred_trans(dc1394camera_t *camera,
                               dc1394bool_t HoldImage, dc1394bool_t FastCapture,
                               uint32_t FifoSize, uint32_t NumOfImages,
@@ -1321,7 +1322,7 @@
 /************************************************************************/
 /* Get GPData info                                                        */
 /************************************************************************/
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_avt_get_gpdata_info(dc1394camera_t *camera, uint32_t *BufferSize)
 {
     dc1394error_t err;
diff -ur libdc1394-2.1.3/configure.in //home/portage/portage/media-libs/libdc1394-2.1.3/work/libdc1394-2.1.3/configure.in
--- libdc1394-2.1.3/configure.in	2011-01-03 08:12:36.000000000 +0400
+++ libdc1394-2.1.3/configure.in	2011-07-09 20:06:40.000000000 +0400
@@ -107,6 +110,23 @@
 
 AC_FUNC_MMAP
 
+dnl Check for -fvisibility=hidden, when using gcc
+AC_CACHE_CHECK([whether -fvisibility=hidden is supported],[dt_visibility_hidden],
+	       [dt_visibility_hidden=no
+		if test "x$GCC" == "xyes"
+		then
+			AC_LANG(C)
+			AC_LANG_CONFTEST([AC_LANG_PROGRAM([[]],[[]])])
+			if $CC $CPPFLAGS $CFLAGS -fvisibility=hidden conftest.c
+			then
+				dt_visibility_hidden=yes
+			fi
+		fi])
+if test "$dt_visibility_hidden" == "yes"
+then
+	CFLAGS="$CFLAGS -fvisibility=hidden"
+fi
+
 dnl ---------------------------------------------------------------------------
 dnl When making a release:
 dnl  1. If the library source code has changed at all since the last release,
diff -ur libdc1394-2.1.3/dc1394/vendor/basler.c //home/portage/portage/media-libs/libdc1394-2.1.3/work/libdc1394-2.1.3/dc1394/vendor/basler.c
--- libdc1394-2.1.3/dc1394/vendor/basler.c	2009-05-25 10:49:17.000000000 +0400
+++ libdc1394-2.1.3/dc1394/vendor/basler.c	2011-07-09 20:28:54.000000000 +0400
@@ -96,7 +96,7 @@
 /*
  * Tests whether the camera supports Basler SFF
  */
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_basler_sff_is_available (dc1394camera_t* camera, dc1394bool_t *available)
 {
     uint32_t data;
@@ -137,7 +137,7 @@
 /**
  * Tests whether the camera supports the specified SFF feature
  */
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_basler_sff_feature_is_available (dc1394camera_t* camera, dc1394basler_sff_feature_t feature_id, dc1394bool_t *available)
 {
     const sff_feature *feature_desc = NULL;
@@ -177,7 +177,7 @@
 /*
  * enables a specific SFF feature
  */
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_basler_sff_feature_enable (dc1394camera_t* camera, dc1394basler_sff_feature_t feature_id, dc1394switch_t on_off)
 {
     const sff_feature *feature_desc = NULL;
@@ -246,7 +246,7 @@
 /*
  * check if a feature is enabled or not
  */
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_basler_sff_feature_is_enabled (dc1394camera_t* camera, dc1394basler_sff_feature_t feature_id, dc1394bool_t *is_enabled)
 {
     const sff_feature *feature_desc = NULL;
@@ -290,7 +290,7 @@
 /*
  * print a feature
  */
-dc1394error_t
+DLL_PUBLIC dc1394error_t
 dc1394_basler_sff_feature_print (dc1394camera_t* camera, dc1394basler_sff_feature_t feature_id, FILE *fd)
 {
     dc1394error_t err;
@@ -355,7 +355,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394error_t dc1394_basler_sff_feature_print_all (dc1394camera_t* camera, FILE *fd)
+DLL_PUBLIC dc1394error_t dc1394_basler_sff_feature_print_all (dc1394camera_t* camera, FILE *fd)
 {
     uint32_t i = DC1394_BASLER_SFF_FEATURE_MIN;
     while (i < DC1394_BASLER_SFF_FEATURE_MAX) {
@@ -366,7 +366,7 @@
     return DC1394_SUCCESS;
 }
 
-dc1394bool_t dc1394_basler_sff_check_crc (const uint8_t* frame_buffer, uint32_t frame_size)
+DLL_PUBLIC dc1394bool_t dc1394_basler_sff_check_crc (const uint8_t* frame_buffer, uint32_t frame_size)
 {
     uint32_t current_crc, desired_crc;
 
@@ -381,7 +381,7 @@
 /*
  * Initializes the struct for iterating
  */
-dc1394error_t dc1394_basler_sff_chunk_iterate_init (dc1394basler_sff_t* chunk, void *frame_buffer, uint32_t frame_size, dc1394bool_t has_crc_checksum)
+DLL_PUBLIC dc1394error_t dc1394_basler_sff_chunk_iterate_init (dc1394basler_sff_t* chunk, void *frame_buffer, uint32_t frame_size, dc1394bool_t has_crc_checksum)
 {
     if (chunk == NULL || frame_buffer == NULL || frame_size == 0)
         return DC1394_FAILURE;
@@ -403,7 +403,7 @@
 /**
  * Iterates over the available SFF chunks in the frame buffer
  */
-dc1394error_t dc1394_basler_sff_chunk_iterate (dc1394basler_sff_t* chunk)
+DLL_PUBLIC dc1394error_t dc1394_basler_sff_chunk_iterate (dc1394basler_sff_t* chunk)
 {
     dc1394basler_sff_chunk_tail_t* tail;
     const sff_feature* feature_desc;
@@ -446,7 +446,7 @@
 /**
  * Finds a specific SFF chunk in the frame buffer
  */
-dc1394error_t dc1394_basler_sff_chunk_find (dc1394basler_sff_feature_t feature_id, void** chunk_data, void* frame_buffer, uint32_t frame_size, dc1394bool_t has_crc_checksum)
+DLL_PUBLIC dc1394error_t dc1394_basler_sff_chunk_find (dc1394basler_sff_feature_t feature_id, void** chunk_data, void* frame_buffer, uint32_t frame_size, dc1394bool_t has_crc_checksum)
 {
     dc1394basler_sff_t chunk;
     dc1394bool_t found = DC1394_FALSE;
