commit 6c2024be59f7d408461f66f47cedfb6c854add74
Author: Sebastian Trueg <trueg@kde.org>
Date:   Sat Nov 5 13:36:02 2011 +0100

    Initialize m_classAndPropertyTree to 0.
    
    BUG: 285789
    FIXED-IN: 4.7.4

diff --git a/nepomuk/services/storage/repository.cpp b/nepomuk/services/storage/repository.cpp
index 784cbe9..50c4cb7 100644
--- a/nepomuk/services/storage/repository.cpp
+++ b/nepomuk/services/storage/repository.cpp
@@ -60,6 +60,7 @@ Nepomuk::Repository::Repository( const QString& name )
     : m_name( name ),
       m_state( CLOSED ),
       m_model( 0 ),
+      m_classAndPropertyTree( 0 ),
       m_inferencer( 0 ),
       m_removableStorageModel( 0 ),
       m_dataManagementModel( 0 ),
